packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -112,176 +102,140 @@ const propTypes = {
-  }
-  if (rejectAction) {
-    rejectButton = (emphasizedAction === 'reject')
-      ? <Button text={rejectAction.text} variant={ButtonVariants.EMPHASIS} onClick={rejectAction.onClick} />
-      : <Button text={rejectAction.text} onClick={rejectAction.onClick} />;
+    const buttonVariant = emphasizedAction === 'accept' ? { variant: ButtonVariants.EMPHASIS } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="accept" key="accept" text={acceptAction.text} onClick={acceptAction.onClick} />);
   }
 
-  if (buttonOrder === 'rejectFirst') {
-    return (
-      <div className={cx('actions')}>
-        {rejectButton}
-        {acceptButton}
-      </div>
-    );
+  if (rejectAction) {
+    const buttonVariant = emphasizedAction === 'reject' ? { variant: ButtonVariants.EMPHASIS } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="reject" key="reject" text={rejectAction.text} onClick={rejectAction.onClick} />);
   }
 
   return (
     <div className={cx('actions')}>
-      {acceptButton}
-      {rejectButton}
+      {buttonOrder === 'acceptFirst' ? actionButtons : actionButtons.reverse()}
     </div>
   );
 };
 
-const getIcon = (variant, customIcon = null) => {
-  switch (variant) {
-    case variants.ALERT:
-      return (<span className={cx(['icon', 'alert'])} />);
-    case variants.ERROR:
-      return (<span className={cx(['icon', 'error'])} />);
-    case variants.WARNING:
-      return (<span className={cx(['icon', 'warning'])} />);
-    case variants.INFO:
-      return (<span className={cx(['icon', 'info'])} />);
-    case variants.SUCCESS:
-      return (<span className={cx(['icon', 'success'])} />);
-    case variants.CUSTOM:
-      return customIcon;
-    default:
-      return null;
-  }
-};
-
-class NotificationDialog extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.handleKeydown = this.handleKeydown.bind(this);
-    this.escapeKey = KeyCode.KEY_ESCAPE;
-  }
-
-  componentDidMount() {
-    document.addEventListener('keydown', this.handleKeydown);
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('keydown', this.handleKeydown);
-  }
-
-  handleKeydown(e) {
-    const notificationDialog = document.querySelector('[data-terra-notification-dialog]');
-
-    if (e.keyCode === this.escapeKey) {
-      if (notificationDialog) {
-        if (e.target === notificationDialog || notificationDialog.contains(e.target)) {
-          e.stopImmediatePropagation();
+const NotificationDialog = (props) => {
+  const theme = React.useContext(ThemeContext);
+  useEffect(() => {
+    function handleKeydown(e) {
+      const notificationDialog = document.querySelector('[data-terra-notification-dialog]');
+
+      if (e.keyCode === KEY_ESCAPE) {
+        if (notificationDialog) {
+          if (e.target === notificationDialog || notificationDialog.contains(e.target)) {
+            e.stopImmediatePropagation();
```

Comment:
Is this logic to intercept the Escape key necessary? I would have thought that the AbstractModal's closeOnEsc prop (as false) would be enough for us to get the behavior we want. But maybe this is doing something else?

Reply:
Good question. I didn't really mess with it before your comment, but from what I can tell removing this still works as we expect. I updated to remove it: https://github.com/cerner/terra-framework/pull/1105/commits/710f82574445d978a9d549e57f97b2bdedef804e#diff-cb62dc7ce0197bc569a9952450bb9fe5L140